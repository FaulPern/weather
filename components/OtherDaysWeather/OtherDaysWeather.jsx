// Retourne array de [length] nombre de jours commençant par le jour "day"
function forecastCalculator (day, length) {
  const forecastDays = []
  for (let index = 0; index < length; index++) {
    if (day <= 6) {
      forecastDays.push(day)
      day++
    } else {
      day = 0
      forecastDays.push(day)
      day++
    }
  }
  return forecastDays
}

const OtherDaysWeather = props => {
  const date = new Date()
  const currentDay = date.getDay()
  const dayName = ['Dim', 'Lun', 'Mar', 'Mer', 'Jeu', 'Ven', 'Sam']
  const forecastDays = forecastCalculator(currentDay, 5)
  const forecast = { ...props.data }
  return (
    <ul className='grid'>
      {forecastDays.map(day => (
        <li key={day.toString()}>
          <h2>{dayName[day]}</h2>
          <span>{forecast[day].tempmin}° / {forecast[day].tempmax}°</span>
        </li>
      ))}
    </ul>
  )
}

export default OtherDaysWeather
