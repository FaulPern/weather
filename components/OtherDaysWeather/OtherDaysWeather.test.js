import { mount } from 'enzyme'
import montreux from '../../mocks/cities/montreux.json'
import OtherDaysWeather from './OtherDaysWeather'

describe('OtherDaysWeather component', () => {
  const mountComponent = mount(<OtherDaysWeather data={montreux} />)
  it('should render', () => {
    expect(mountComponent).toMatchSnapshot()
  })
})
