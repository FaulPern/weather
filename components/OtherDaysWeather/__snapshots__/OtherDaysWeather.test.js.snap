// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherDaysWeather component should render 1`] = `
<OtherDaysWeather
  data={
    Object {
      "current": Object {},
      "forecast": Object {
        "0": Object {
          "hourly": Object {
            "00:00": Object {
              "condition": "sun",
              "temp": 14,
            },
            "03:00": Object {
              "condition": "rain",
              "temp": 16,
            },
            "06:00": Object {
              "condition": "rain",
              "temp": 15,
            },
            "09:00": Object {
              "condition": "rain",
              "temp": 14,
            },
            "12:00": Object {
              "condition": "sun",
              "temp": 22,
            },
            "15:00": Object {
              "condition": "sun",
              "temp": 24,
            },
            "18:00": Object {
              "condition": "sun",
              "temp": 13,
            },
            "21:00": Object {
              "condition": "rain",
              "temp": 17,
            },
          },
          "tempavg": "18",
          "tempmax": "25",
          "tempmin": "11",
        },
        "1": Object {
          "hourly": Object {
            "00:00": Object {
              "condition": "cloud",
              "temp": 11,
            },
            "03:00": Object {
              "condition": "sun",
              "temp": 15,
            },
            "06:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "09:00": Object {
              "condition": "cloud",
              "temp": 14,
            },
            "12:00": Object {
              "condition": "sun",
              "temp": 22,
            },
            "15:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "18:00": Object {
              "condition": "cloud",
              "temp": 16,
            },
            "21:00": Object {
              "condition": "sun",
              "temp": 17,
            },
          },
          "tempavg": "18",
          "tempmax": "25",
          "tempmin": "11",
        },
        "2": Object {
          "hourly": Object {
            "00:00": Object {
              "condition": "rain",
              "temp": 15,
            },
            "03:00": Object {
              "condition": "sun",
              "temp": 15,
            },
            "06:00": Object {
              "condition": "rain",
              "temp": 14,
            },
            "09:00": Object {
              "condition": "sun",
              "temp": 19,
            },
            "12:00": Object {
              "condition": "cloud",
              "temp": 21,
            },
            "15:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "18:00": Object {
              "condition": "sun",
              "temp": 16,
            },
            "21:00": Object {
              "condition": "rain",
              "temp": 14,
            },
          },
          "tempavg": "18",
          "tempmax": "25",
          "tempmin": "11",
        },
        "3": Object {
          "hourly": Object {
            "00:00": Object {
              "condition": "cloud",
              "temp": 13,
            },
            "03:00": Object {
              "condition": "rain",
              "temp": 18,
            },
            "06:00": Object {
              "condition": "sun",
              "temp": 19,
            },
            "09:00": Object {
              "condition": "rain",
              "temp": 19,
            },
            "12:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "15:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "18:00": Object {
              "condition": "rain",
              "temp": 15,
            },
            "21:00": Object {
              "condition": "rain",
              "temp": 18,
            },
          },
          "tempavg": "18",
          "tempmax": "25",
          "tempmin": "11",
        },
        "4": Object {
          "hourly": Object {
            "00:00": Object {
              "condition": "cloud",
              "temp": 13,
            },
            "03:00": Object {
              "condition": "rain",
              "temp": 18,
            },
            "06:00": Object {
              "condition": "sun",
              "temp": 19,
            },
            "09:00": Object {
              "condition": "rain",
              "temp": 19,
            },
            "12:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "15:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "18:00": Object {
              "condition": "rain",
              "temp": 15,
            },
            "21:00": Object {
              "condition": "rain",
              "temp": 18,
            },
          },
          "tempavg": "18",
          "tempmax": "25",
          "tempmin": "11",
        },
        "5": Object {
          "hourly": Object {
            "00:00": Object {
              "condition": "cloud",
              "temp": 13,
            },
            "03:00": Object {
              "condition": "rain",
              "temp": 18,
            },
            "06:00": Object {
              "condition": "sun",
              "temp": 19,
            },
            "09:00": Object {
              "condition": "rain",
              "temp": 19,
            },
            "12:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "15:00": Object {
              "condition": "cloud",
              "temp": 20,
            },
            "18:00": Object {
              "condition": "rain",
              "temp": 15,
            },
            "21:00": Object {
              "condition": "rain",
              "temp": 18,
            },
          },
          "tempavg": "18",
          "tempmax": "25",
          "tempmin": "11",
        },
        "6": Object {
          "hourly": Object {
            "00:00": Object {
              "condition": "rain",
              "temp": 16,
            },
            "03:00": Object {
              "condition": "sun",
              "temp": 14,
            },
            "06:00": Object {
              "condition": "sun",
              "temp": 16,
            },
            "09:00": Object {
              "condition": "sun",
              "temp": 17,
            },
            "12:00": Object {
              "condition": "sun",
              "temp": 23,
            },
            "15:00": Object {
              "condition": "sun",
              "temp": 24,
            },
            "18:00": Object {
              "condition": "sun",
              "temp": 13,
            },
            "21:00": Object {
              "condition": "sun",
              "temp": 14,
            },
          },
          "tempavg": "18",
          "tempmax": "25",
          "tempmin": "11",
        },
      },
      "name": "Montreux",
      "slug": "montreux",
    }
  }
>
  <ul
    className="grid"
  >
    <li
      key="5"
    >
      <h2>
        Ven
      </h2>
      <span>
        11
        ° / 
        25
        °
      </span>
    </li>
    <li
      key="6"
    >
      <h2>
        Sam
      </h2>
      <span>
        11
        ° / 
        25
        °
      </span>
    </li>
    <li
      key="0"
    >
      <h2>
        Dim
      </h2>
      <span>
        11
        ° / 
        25
        °
      </span>
    </li>
    <li
      key="1"
    >
      <h2>
        Lun
      </h2>
      <span>
        11
        ° / 
        25
        °
      </span>
    </li>
    <li
      key="2"
    >
      <h2>
        Mar
      </h2>
      <span>
        11
        ° / 
        25
        °
      </span>
    </li>
  </ul>
</OtherDaysWeather>
`;
